### ISSUE TITLE
Example of as_pairlist() shows an incorrect pairlist

### ISSUE THREAD

**Issue by yutannihilation** [extendr/extendr#191]
(I expect this isn't a very big problem. Just a note not to forget to review this again when ongoing changes complete...)

The document uses `na_str()` as an argument name, and the user probably expect this means an unnamed argument, but in actual, it's not.

<https://extendr.github.io/extendr/extendr_api/robj/enum.Robj.html#method.as_pairlist>:
``` rust
use extendr_api::prelude::*;
test! {
    let names_and_values = vec![("a", r!(1)), ("b", r!(2)), (na_str(), r!(3))];
    let pairlist = Pairlist{ names_and_values };
    let robj = r!(pairlist.clone());
    assert_eq!(robj.as_pairlist().unwrap(), pairlist);
}
```

``` r
rextendr::rust_function(r"(
fn call_pairlist(f: Robj) -> Robj {
  let names_and_values = vec![("a", r!(1)), ("b", r!(2)), (na_str(), r!(3))];
  let pairlist = Pairlist{ names_and_values };
  f.call(pairlist.into_robj()).unwrap()
}
)", patch.crates_io = list(`extendr-api` = list(git = "https://github.com/extendr/extendr"))
)
#> build directory: /tmp/Rtmp3Tjm0Q/file1022ad72ffb210
#> Writting wrappers to:
#> /tmp/Rtmp3Tjm0Q/file1022ad72ffb210/target/extendr_wrappers.R

call_pairlist(function(...) list(...))
#> $a
#> [1] 1
#> 
#> $b
#> [1] 2
#> 
#> $`NA`
#> [1] 3
```

<sup>Created on 2021-03-11 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by yutannihilation** [comment]
This might be related. Why doesn't this work...?

``` r
rextendr::rust_function(
  r"--(
  fn f() -> Robj {
    let l = pairlist!(1, 2);
    l.set_names(vec!["".to_string(), "a".to_string()]);
    l
  }
  )--",
  patch.crates_io = list("extendr-api" = list(git = "https://github.com/extendr/extendr"))
)
#> build directory: /tmp/Rtmp9cZbrs/file10b294127564b4
#> Writing wrappers to:
#> /tmp/Rtmp9cZbrs/file10b294127564b4/target/extendr_wrappers.R

f()
#> [[1]]
#> [1] 1
#> 
#> $a
#> [1] 2

rextendr::rust_function(
  r"--(
  fn f2() -> Robj {
    let l = Pairlist { names_and_values: vec![("", r!(1)), ("a", r!(2))] };
    l.into_robj()
  }
  )--",
  patch.crates_io = list("extendr-api" = list(git = "https://github.com/extendr/extendr"))
)
#> build directory: /tmp/Rtmp9cZbrs/file10b294127564b4
#> 
#> Writing wrappers to:
#> /tmp/Rtmp9cZbrs/file10b294127564b4/target/extendr_wrappers.R

f2()
#> Error in f2(): attempt to use zero-length variable name
```

<sup>Created on 2021-03-11 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by yutannihilation** [comment]
Fixed by #190, but let's continue the discussion on #195
